frontend/src/scenes/data-management/definition/definitionLogic.ts

```diff
@@ -85,6 +87,20 @@ export const definitionLogic = kea<definitionLogicType>([
 
                     return definition
                 },
+                deleteDefinition: async () => {
+                    if (values.isEvent) {
+                        await api.eventDefinitions.delete({ eventDefinitionId: values.definition.id })
+                    } else {
+                        await api.propertyDefinitions.delete({ propertyDefinitionId: values.definition.id })
+                    }
+                    router.actions.push(values.isEvent ? urls.eventDefinitions() : urls.propertyDefinitions())
+                    if (values.isEvent) {
+                        eventDefinitionsTableLogic.findMounted()?.actions.loadEventDefinitions()
+                    } else {
+                        propertyDefinitionsTableLogic.findMounted()?.actions.loadPropertyDefinitions()
+                    }
```

Comment:
don't we in other places where we do something similar just update the frontend local data (e.g. person property updates update the local data by adding / removing the property, which is more efficient than reloading everything - nicer experience if you're deleting many things and reloading is slow)?

Reply:
We do, I actually wanted to do that first here… But the event/property definition table logics have some custom-ish pagination handling (e.g. there's a next/previous page URL normalization process), which I don't fully understand, so just reloading completely to make sure there's no stale data – which is worse than having to make an extra request.